feat: добавлен функционал по работе с пользователями и вещами

Beablad opened the pull request.

Commit:
feat: добавлен функционал по работе с пользователями и вещами

Commit:
fix: добавлены и убраны пробелы

Commit:
fix: добавлены и убраны пробелы

Commit:
fix: добавлены и убраны пробелы

Troitsa commented on src/main/java/ru/practicum/shareit/exception/ExceptionHandle.java:
```diff
@@ -0,0 +1,39 @@
+    public Map<String, String> handleNoAccessException(NoAccessException e) {
+        return Map.of("error", "Передан неверный идентификатор",
+                "errorMessage", e.getMessage());
+    }
```
Нужно еще добавить хендлер на все остальные эксепшены, примерно так:

    @ExceptionHandler(Exception.class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    public String handleException(Throwable exception) {
        log.error(exception.getMessage());
        return new ErrorResponse(e.getMessage());
    }

Troitsa commented on src/main/java/ru/practicum/shareit/exception/ExceptionHandle.java:
```diff
@@ -0,0 +1,39 @@
+    @ExceptionHandler
+    @ResponseStatus(HttpStatus.CONFLICT)
+    public Map<String, String> handleConflictException(ConflictException e) {
+        return Map.of("error", "Передан неверный идентификатор",
```
можно добавить логгирование

Troitsa commented on src/main/java/ru/practicum/shareit/item/dto/ItemMapper.java:
```diff
@@ -0,0 +1,37 @@
+/**
+ * TODO Sprint add-item-requests.
+ */
+@FieldDefaults(level = AccessLevel.PRIVATE)
```
полей тут нет, можно не использовать аннотацию

Troitsa commented on src/main/java/ru/practicum/shareit/item/repository/ItemRepository.java:
```diff
@@ -0,0 +1,20 @@
+
+public interface ItemRepository {
+
+    public Item createItem(Item item, User owner);
```
Методы интерфейса неявно публичны, поскольку смысл интерфейса в том, чтобы объявить методы, которые могут видеть другие классы. Поэтому использование модификаторов в интерфейсе - излишне

Commit:
fix: убраны модификаторы доступа во всех интерфейсах
feat: добавлено логирование в ExceptionHandle, а также метод для отлова остальных ошибок

Commit:
fix: убраны модификаторы доступа во всех интерфейсах
feat: добавлено логирование в ExceptionHandle, а также метод для отлова остальных ошибок

Commit:
fix: убраны модификаторы доступа во всех интерфейсах
feat: добавлено логирование в ExceptionHandle, а также метод для отлова остальных ошибок
